src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java

```diff
@@ -41,12 +49,47 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse submitSuccessfulMediation(CallbackParams callbackParams) {
+        List<Element<ManageDocument>> updatedManageDocumentsList =
+            updateManageDocumentsListWithMediationAgreementDocument(callbackParams);
+
         CaseData caseDataUpdated = callbackParams.getCaseData().toBuilder()
             .businessProcess(BusinessProcess.ready(MEDIATION_SUCCESSFUL))
+            .manageDocuments(updatedManageDocumentsList)
             .build();
+
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDataUpdated.toMap(objectMapper))
             .state(CASE_STAYED.name())
             .build();
     }
+
+    private List<Element<ManageDocument>> updateManageDocumentsListWithMediationAgreementDocument(
+        CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<Element<ManageDocument>> manageDocumentsList =
+            Optional.ofNullable(caseData.getManageDocuments()).orElse(new ArrayList<>());
+        Optional<MediationAgreementDocument> mediationAgreementDocument = Optional.ofNullable(caseData)
+            .map(data -> data.getMediation())
+            .map(mediation -> mediation.getMediationSuccessful())
+            .map(successful -> successful.getMediationAgreement());
+
+        mediationAgreementDocument.ifPresent(document -> {
+            ManageDocument manageDocument = ManageDocument.builder()
+                .documentLink(document.getDocument())
+                .documentName(document.getName())
+                .documentType(mapDocumentTypeToManageDocumentType(document.getDocumentType()))
+                .build();
+            manageDocumentsList.add(element(manageDocument));
+        });
+        return manageDocumentsList;
+    }
+
+    private ManageDocumentType mapDocumentTypeToManageDocumentType(DocumentType documentType) {
```

Comment:
this mapping should not be inside the handler. what if this knowledge is needed else where? hanlder should not have a lot of business logic. think about it as a controller. perhaps the mapping should happen in the ManageDocumentType enum?

Reply:
moved to a util class